src/ir_Teco.h

```diff
@@ -0,0 +1,135 @@
+// Copyright 2019 Fabien Valthier
+
+#ifndef IR_TECO_H_
+#define IR_TECO_H_
+
+#ifndef UNIT_TEST
+#include <Arduino.h>
+#else
+#include <string>
+#endif
+#include "IRremoteESP8266.h"
+#include "IRsend.h"
+
+// Constants. Using LSB to be able to send only 35bits.
+const uint8_t kTecoAuto = 0;  // 0b000
+const uint8_t kTecoCool = 1;  // 0b001
+const uint8_t kTecoDry = 2;  // 0b010
+const uint8_t kTecoFan = 3;  // 0b110
+const uint8_t kTecoHeat = 4;  // 0b001
+const uint8_t kTecoFanAuto = 0;  // 0b00
+const uint8_t kTecoFanHigh = 3;  // 0b11
+const uint8_t kTecoFanMed = 2;  // 0b10
+const uint8_t kTecoFanLow = 1;  // 0b01
+const uint8_t kTecoMinTemp = 16;   // 16C
+const uint8_t kTecoMaxTemp = 30;   // 30C
+
+const uint64_t kTecoModeMask =   0b00000000000000000000000000000000111;
+const uint64_t kTecoPower =      0b00000000000000000000000000000001000;
+const uint64_t kTecoFanMask =    0b00000000000000000000000000000110000;
+const uint64_t kTecoSwing =      0b00000000000000000000000000001000000;
+const uint64_t kTecoSleep =      0b00000000000000000000000000010000000;
+const uint64_t kTecoTempMask =   0b00000000000000000000000111100000000;
+const uint64_t kTecoTimerHalfH = 0b00000000000000000000001000000000000;
+const uint64_t kTecoTimerTenHr = 0b00000000000000000000110000000000000;
+const uint64_t kTecoTimerOn =    0b00000000000000000001000000000000000;
+const uint64_t kTecoTimerUniHr = 0b00000000000000011110000000000000000;
+const uint64_t kTecoReset =      0b01001010000000000000010000000000000;
```

Comment:
Is there any particular reason why you changed the Reset state to have the timer at 10 hours? (although the timer is not on).
I would have put it at: 0b01001010000000000000000000000000000

Reply:
Yes, if you look here: https://github.com/markszabo/IRremoteESP8266/pull/622/commits/cd0d60899afb7d261ff0e3bc84653e159c86f9b3#diff-47417b72c15b4b042cc8d367a2a69633R240 in this unit test, I try to construct the message code you added via the `rawData` unit test.
As we don't _yet_ have the timer functions, I couldn't set that bit _yet_ via published/existing methods. I did have a look at implementing them, but I couldn't understand your comments well enough to do so. i.e. the comments describing the time/hour structure.. e.g. it wasn't clear to me.